src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -65,4 +72,42 @@ public static void deleteTempTargetS3Bucket() throws IOException {
     BeforeActions.scenario.write("Deleted S3 Target Bucket " + s3TargetBucketName);
     s3TargetBucketName = StringUtils.EMPTY;
   }
+
+  public static void verifyDataIsTransferredToTargetS3Bucket() {
+    String s3Bucket = TestSetupHooks.s3TargetBucketName;
+    boolean flag = false;
+    StringBuilder sb = new StringBuilder();
+    try {
+      for (S3ObjectSummary s3ObjectSummary : S3Client.listObjects(s3Bucket)) {
+        if (s3ObjectSummary.getKey().contains("part")) {
+          flag = true;
+          String content = S3Client.getObject(s3Bucket, s3ObjectSummary.getKey());
+          sb.append(content);
+        }
+      }
+      if (flag) {
+        String path = Paths.get(Objects.requireNonNull(TestSetupHooks.class.getResource
+          ("/" + PluginPropertyUtils.pluginProp("outputCSV"))).getPath()).toString();
+        Assert.assertEquals("Output content should match", readFile(path), sb.toString());
+        BeforeActions.scenario.write("Data transferred to s3 bucket " + s3Bucket + " successfully");
+      } else {
+        Assert.fail("Data not transferred to target s3 bucket " + s3Bucket);
+      }
+    } catch (Exception e) {
+      if (e.getMessage().contains("The specified bucket does not exist")) {
+        Assert.fail("Target s3 bucket " + s3Bucket + " not created - " + e.getMessage());
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  private static String readFile(String path) throws IOException {
```

Comment:
This is used multiple times for other e2e tests. Plan to move this logic to cdap-e2e repo